projects/api-client/src/api/build.service.ts

```diff
@@ -140,24 +151,108 @@ export class BuildService {
+     * @param match Filter by matching on any supported fields.
      * @param observe set whether or not to return the data Observable as the body, response or events. defaults to returning the body.
      * @param reportProgress flag to report request and response progress.
      */
-    public buildBuildidLogPost(buildid: number, data: MainBuildLog, observe?: 'body', reportProgress?: boolean): Observable<any>;
-    public buildBuildidLogPost(buildid: number, data: MainBuildLog, observe?: 'response', reportProgress?: boolean): Observable<HttpResponse<any>>;
-    public buildBuildidLogPost(buildid: number, data: MainBuildLog, observe?: 'events', reportProgress?: boolean): Observable<HttpEvent<any>>;
-    public buildBuildidLogPost(buildid: number, data: MainBuildLog, observe: any = 'body', reportProgress: boolean = false ): Observable<any> {
+    public getBuildList(limit?: number, offset?: number, orderby?: Array<string>, projectId?: number, scheduledAfter?: Date, scheduledBefore?: Date, finishedAfter?: Date, finishedBefore?: Date, environment?: string, gitBranch?: string, stage?: string, isInvalid?: boolean, status?: 'Scheduling' | 'Running' | 'Completed' | 'Failed', statusId?: number, environmentMatch?: string, gitBranchMatch?: string, stageMatch?: string, match?: string, observe?: 'body', reportProgress?: boolean): Observable<ResponsePaginatedBuilds>;
+    public getBuildList(limit?: number, offset?: number, orderby?: Array<string>, projectId?: number, scheduledAfter?: Date, scheduledBefore?: Date, finishedAfter?: Date, finishedBefore?: Date, environment?: string, gitBranch?: string, stage?: string, isInvalid?: boolean, status?: 'Scheduling' | 'Running' | 'Completed' | 'Failed', statusId?: number, environmentMatch?: string, gitBranchMatch?: string, stageMatch?: string, match?: string, observe?: 'response', reportProgress?: boolean): Observable<HttpResponse<ResponsePaginatedBuilds>>;
+    public getBuildList(limit?: number, offset?: number, orderby?: Array<string>, projectId?: number, scheduledAfter?: Date, scheduledBefore?: Date, finishedAfter?: Date, finishedBefore?: Date, environment?: string, gitBranch?: string, stage?: string, isInvalid?: boolean, status?: 'Scheduling' | 'Running' | 'Completed' | 'Failed', statusId?: number, environmentMatch?: string, gitBranchMatch?: string, stageMatch?: string, match?: string, observe?: 'events', reportProgress?: boolean): Observable<HttpEvent<ResponsePaginatedBuilds>>;
+    public getBuildList(limit?: number, offset?: number, orderby?: Array<string>, projectId?: number, scheduledAfter?: Date, scheduledBefore?: Date, finishedAfter?: Date, finishedBefore?: Date, environment?: string, gitBranch?: string, stage?: string, isInvalid?: boolean, status?: 'Scheduling' | 'Running' | 'Completed' | 'Failed', statusId?: number, environmentMatch?: string, gitBranchMatch?: string, stageMatch?: string, match?: string, observe: any = 'body', reportProgress: boolean = false ): Observable<any> {
```

Comment:
Fuck, this endpoint is a mess. I was afraid of this.

Would like if the generated function accepted an object or interface with all the fields instead of every parameter as separate function parameters. Especially as JS/TS doesn't support named arguments to allow calling with only some optional parameters

Reply:
Neither version 2 (https://github.com/swagger-api/swagger-codegen/blob/2ff2459623bcea73d5fbb0cbf3be0bb5623795c5/modules/swagger-codegen/src/main/resources/typescript-angular/api.service.mustache#L156-L179) nor version 3 (https://github.com/swagger-api/swagger-codegen-generators/blob/51fab5f9cab64bf8221ad323cd3e16e230e0796d/src/main/resources/handlebars/typescript-angular/api.service.mustache#L159-L182) seems to support that kind of use case.

Do we maybe want to consider a different tool? Such as https://github.com/jnwltr/swagger-angular-generator ? Seems to generate objects for it: https://github.com/jnwltr/swagger-angular-generator/blob/398f221b9a08f1597566472e4a5df42a0dcaea28/demo-app/client/generated/controllers/Params.ts

Other alternatives would be:
- override the generated code with an inheriting type that merely overrides the method with one that takes in object instead
- send a patch towards the swagger codegen repo to start supporting params object

Thoughts? Might need a verbal discussion

Reply:
I think both 1 and 2 together is the way to go.
1 especially is easy, effective, and a minor (currently!) pain if we change the function(s). Definitely better than living with it though I think; it very quickly gets messy and confusing regarding what is what when calling it.